src/aiodynamo/utils.py

```diff
@@ -197,3 +201,43 @@ def deparametetrize(
     for key, value in params.names.items():
         expression = expression.replace(key, value)
     return expression
+
+
+@dataclass(frozen=True)
+class MinLen2AppendOnlyList(Generic[T]):
+    first: T
+    second: T
+    rest: tuple[T, ...]
+
+    @classmethod
+    def create(cls, first: T, second: T, *rest: T) -> MinLen2AppendOnlyList[T]:
+        return cls(first, second, rest)
+
+    def appending(self, value: T) -> MinLen2AppendOnlyList[T]:
+        return MinLen2AppendOnlyList(self.first, self.second, (*self.rest, value))
+
+    def extending(self, values: Iterable[T]) -> MinLen2AppendOnlyList[T]:
+        return MinLen2AppendOnlyList(self.first, self.second, (*self.rest, *values))
+
+    def __contains__(self, item: Any) -> bool:
+        return item == self.first or item == self.second or item in self.rest
```

Comment:
Minor (probably goes away if refactored):
https://docs.python.org/3/reference/expressions.html#membership-test-operations
`any(x is e or x == e for e in y)`

Reply:
honestly, most of these magic methods could (should?) probably removed. I just made it as 'sequene-like' as possible for no real reason 😅